service/mantle/ShipEngineServices.xml

```diff
@@ -0,0 +1,445 @@
+                </entity-find-one>
+                <set field="number" from="originTelecomNumber.areaCode+'-'+originTelecomNumber.contactNumber"/>
+                <set field="shipFrom" from="[name:originAddress.toName,phone:number,address_line1:originAddress.address1,
+                       city_locality: originAddress.city, state_province:originState.geoCodeAlpha2,postal_code:originAddress.postalCode,
+                       country_code:originCountry.geoCodeAlpha2,address_residential_indicator:indicatorFrom]"/>
+            </then><else>
+                <message error="true">No origin address set on Shipment Route Segment ${shipmentId}:${shipmentRouteSegmentSeqId}</message>
+            </else></if>
+
+
+            <!-- ==============  Packages  ============== -->
+
+            <entity-find-one entity-name="mantle.shipment.ShipmentPackage" value-field="shipmentPackage">
+                <field-map field-name="shipmentId"/>
+                <field-map field-name="shipmentPackageSeqId" />
+            </entity-find-one>
+            <if condition="!shipmentPackage.weightUomId"><log level="warn" message="This shipmentId ===${shipmentId}=== has no Shipment package, not getting rates"/><return/></if>
+            <entity-find-one entity-name="moqui.basic.UomConversion" value-field="conversion">
+                <field-map field-name="uomId" from="shipmentPackage.weightUomId"/>
+                <field-map field-name="toUomId" value="WT_oz"/>
+            </entity-find-one>
+            <set field="convertedWeight" from="shipmentPackage.weight*conversion.conversionFactor"/>
+            <if condition="!convertedWeight"><log level="warn" message="This ShipmentId ${shipmentId} has no weight in ShipmentPackage, not getting rates"/><return/></if>
+            <set field="weight" from="[value: convertedWeight, unit: 'ounce']"/>
+
+            <set field="packageList" from="[weight:weight]"/>
+            <set field="packages" from="[packageList]"/>
+            <set field="shipment" from="[validateAddress:'noValidation',ship_to:shipTo,ship_from:shipFrom,packages:packages]"/>
+            <set field="rateOptions" from="[carrier_ids:carrierIds,service_codes:serviceCode]"/>
+            <set field="requestMap" from= "[rate_options:rateOptions,shipment:shipment]"/>
+            <set field="responseMap" from="[]"/>
+            <set field="errMsg" type="String"/>
+
+            <script><![CDATA[
+                    org.moqui.util.RestClient restClient = ec.service.rest().method(org.moqui.util.RestClient.POST)
+                    .addHeader("API-Key", "${apiToken}")
+                    .addHeader("Content-Type", "application/json").jsonObject(requestMap)
+                    restClient.uri().protocol("https").host("api.shipengine.com").port(443).path("v1/rates").build()
+                    org.moqui.util.RestClient.RestResponse restResponse = restClient.call()
+                    if (restResponse.statusCode < 200 || restResponse.statusCode >= 300) {
```

Comment:
Replace with,
`
if (restResponse.statusCode < 200 || restResponse.statusCode >= 300) {
    ec.logger.warn("Unsuccessful with status code: ${statusCode} and response: ${response}")
    return
}
`

Reply:
Resolved Issues successfully.